File: src/gestorAplicacion/Consulta.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package gestorAplicacion;

import gestorAplicacion.personas.*;
import java.time.*;

/**
 *
 * @author Roger Vera
 */
public abstract class Consulta {
    protected int id;
    protected Paciente paciente;
    protected Medico medico;
    protected Consultorio consultorio;
    protected LocalDateTime fecha;
    protected Pago pago;
    
    public Consulta(int id, Paciente paciente, Medico medico, Consultorio consultorio, LocalDateTime fecha, Pago pago) {
    	this.id= id;
    	this.paciente = paciente;
    	this.medico = medico;
    	this.consultorio = consultorio;
    	this.fecha = fecha;
    	this.pago = pago;
    }
    
    public Consulta(Paciente paciente, Medico medico, Consultorio consultorio, LocalDateTime fecha, Pago pago) {
    	this.id= (int)(Math.random()*10000+1);
    	this.paciente = paciente;
    	this.medico = medico;
    	this.consultorio = consultorio;
    	this.fecha = fecha;
    	this.pago = pago;
    }
    
    
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Paciente getPaciente() {
        return paciente;
    }

    public void setPaciente(Paciente paciente) {
        this.paciente = paciente;
    }
    public Medico getMedico() {
        return medico;
    }

    public void setMedico(Medico medico) {
        this.medico = medico;
    }

    public Consultorio getConsultorio() {
        return consultorio;
    }

    public void setConsultorio(Consultorio consultorio) {
        this.consultorio = consultorio;
    }

    public LocalDateTime getFecha() {
        return fecha;
    }

    public void setFecha(LocalDateTime fecha) {
        this.fecha = fecha;
    }
    
    public Pago getPago() {
    	return pago;
    }
    
    public void setPago(Pago pago) {
    	this.pago = pago;
    }
}

File: Python/src/gestorAplicacion/Consulta.py
"""
La clase Consulta tiene el fin de ser la clase padre de Cita y Examen agrupando así los atributos y métodos comunes para
estas dos subclases.
"""
import random as rd
class Consulta():
    def __init__(self, id = int(rd.random() * 10000 + 1), paciente = None, medico = None, consultorio = None, 
                 fecha = None, pago = None):
        self._id = id
        self._paciente = paciente
        self._medico = medico
        self._consultorio = consultorio
        self._fecha = fecha
        self._pago = pago
    
    # Getters y Setters
    def getId(self):
        return self._id
    
    def setId(self, id):
        self._id = id
        
    def getPaciente(self):
        return self._paciente
    
    def setPaciente(self, paciente):
        self._paciente = paciente
    
    def getMedico(self):
        return self._medico
    
    def setMedico(self, medico):
        self._medico = medico
        
    def getConsultorio(self):
        return self._consultorio
    
    def setConsultorio(self, consultorio):
        self._consultorio = consultorio
        
    def getFecha(self):
        return self._fecha
    
    def setFecha(self, fecha):
        self._fecha = fecha
    
    def getPago(self):
        return self._pago
    
    def setPago(self, pago):
        self._pago = pago